edi_bank_statement_import/components/process.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021 AGEPoly - Téo Goddet
+# @author: Téo Goddet <teo.goddet@gmail.>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import base64
+
+from odoo import _, api
+from odoo.exceptions import UserError
+
+from odoo.addons.component.core import Component
+
+# TODO: add tests
+class EDIExchangeBankStatementInput(Component):
+    """Process bank statements."""
+
+    _name = "edi.input.bank.statement.process"
+    _inherit = "edi.component.input.mixin"
+    _usage = "input.process.bank.statement"
+
+    def __init__(self, work_context):
+        super().__init__(work_context)
+        self.settings = self.type_settings.get("bank_statement", {})
+
+    def process(self):
+        wiz = self._setup_wizard()
+        result = {
+            "statement_ids": [],
+            "notifications": [],
+        }
+
+        file_data = base64.b64decode(wiz.statement_file)
+        wiz.import_single_file(file_data, result)
+
+        if not result['statement_ids']:
+            self.exchange_record.sudo().message_post(
+                body=_(
+                    "You have already imported this file (%s) or this file "
+                    "only contains already imported transactions."
+                ) % wiz.statement_filename
+            )
+            return True
+
+        if len(result['statement_ids']) == 1:
```

Comment:
pls split this `process` method into smaller methods

Reply:
done